sdk/examples/wallet/accounts.rs

```diff
@@ -1,92 +1,107 @@
-        Ok(account) => account,
-        _ => {
-            wallet
-                .create_account()
-                .with_alias(account_alias.to_string())
-                .finish()
-                .await?
-        }
-    };
+    // Get or create second account
+    let alias2 = var("ACCOUNT_ALIAS_2").unwrap();
+    let account2 = get_or_create_account(&wallet, &alias2).await?;
 
     let accounts = wallet.get_accounts().await?;
+    println!("WALLET ACCOUNTS:");
     for account in accounts {
-        let a = account.details().await;
-        println!("Accounts: {a:#?}");
+        let account = account.details().await;
+        println!("- {}", account.alias());
     }
 
-    let addresses = account.generate_addresses(5, None).await?;
-
-    println!(
-        "{}",
-        request_funds_from_faucet(
-            &std::env::var("FAUCET_URL").unwrap(),
-            &addresses[0].address().to_string()
-        )
-        .await?
-    );
-    tokio::time::sleep(std::time::Duration::from_secs(15)).await;
+    println!("Generating {NUM_ADDRESSES_TO_GENERATE} addresses for account '{alias2}'...");
+    let addresses = account2.generate_addresses(NUM_ADDRESSES_TO_GENERATE, None).await?;
+
+    let balance = account2.sync(None).await?;
+    let funds_before = balance.base_coin().available();
+    println!("Current available funds: {funds_before}");
+
+    println!("Requesting funds from faucet...");
+    let faucet_response =
+        request_funds_from_faucet(&var("FAUCET_URL").unwrap(), &addresses[0].address().to_string()).await?;
+    println!("Response from faucet: {}", faucet_response.trim_end());
+
+    println!("Waiting for funds (timeout=60s)...");
+    // Check for changes to the balance
+    let start = std::time::Instant::now();
+    let balance = loop {
+        if start.elapsed().as_secs() > 60 {
+            println!("Timeout: waiting for funds took too long");
+            return Ok(());
+        };
+        let now = Instant::now();
+        let balance = account2.sync(None).await?;
+        if balance.base_coin().available() > funds_before {
+            println!("Account synced in: {:.2?}", now.elapsed());
+            break balance;
+        } else {
+            tokio::time::sleep(instant::Duration::from_secs(2)).await;
+        }
+    };
 
-    let addresses = account.addresses().await?;
-    println!("Addresses: {}", addresses.len());
+    println!("New available funds: {}", balance.base_coin().available());
 
-    let now = Instant::now();
-    let balance = account.sync(None).await?;
-    println!("Syncing took: {:.2?}", now.elapsed());
-    println!("Balance: {balance:?}");
+    let addresses = account2.addresses().await?;
+    println!("Number of addresses in {alias2}'s account: {}", addresses.len());
+    println!("{alias2}'s base coin balance:\n{:#?}", balance.base_coin());
 
     Ok(())
 }
+
+async fn get_or_create_account(wallet: &Wallet, alias: &str) -> Result<Account> {
```

Comment:
we could use this in more places i think. Maybe even an actual wallet method?

Reply:
Hmm, okay. If you think that could be useful in more places, I think we should create an issue for that?!

Reply:
Can you open an issue for that @kwek20 ?

Reply:
wanna resolve? :trollface:

Reply:
https://github.com/iotaledger/iota-sdk/issues/523